frontend/backend/farcaster/farcaster.ts

```diff
@@ -0,0 +1,27 @@
+import { publishNewUserKeysCast } from "@/lib/api/backend/farcaster";
+import { ERRORS } from "@/lib/errors";
+import prisma from "@/lib/prisma";
+import { SocialProfileType } from "@prisma/client";
+
+export async function publishNewUserCast(privyUserId: string) {
+  if (process.env.NODE_ENV !== "production") {
+    const user = await prisma.user.findUniqueOrThrow({ where: { privyUserId } });
```

Comment:
btw I've noticed this pattern above, but you can use `include: { socialProfile: true}` on this query to not have to run the next query to fetch the farcaster profile, you can also use filters if it's a conditional join